src/integrations/admin/indexing-notification-integration.php

```diff
@@ -147,16 +147,14 @@ public function register_hooks() {
 			\add_action( 'admin_init', [ $this, 'cleanup_notification' ] );
 		}
 
-		if ( $this->options_helper->get( 'indexing_reason' ) || ! \wp_next_scheduled( self::NOTIFICATION_ID ) ) {
-			\add_action( 'admin_init', [ $this, 'create_notification' ] );
-		}
-
 		if ( ! \wp_next_scheduled( self::NOTIFICATION_ID ) ) {
 			\wp_schedule_event( $this->date_helper->current_time(), 'daily', self::NOTIFICATION_ID );
-			return;
+			\add_action( self::NOTIFICATION_ID, [ $this, 'create_notification' ] );
```

Comment:
This `add_action` should definitely be outside of this `if` block. The action will be triggered by scheduled event so if it is triggered then this block will never be reached as it triggering means it has been scheduled.

I also wonder if the `_wpseo_activate` is early enough that the other changes in this file are no longer needed. I would first try reverting the full contents of this file and see simply setting the reason is sufficient.